streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -160,26 +172,190 @@ public void shouldIdempotentlyUpdateSubscriptionFromActiveAssignment() {
+        );
+        final Map<TaskId, Long> expectedOffsetSums = mkMap(mkEntry(taskId00, 15L));
+
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        replay(stateDirectory);
+
+        taskManager.handleRebalanceStart(singleton("topic"));
+        final StateMachineTask restoringTask = handleAssignment(
+            emptyMap(),
+            taskId00Assignment,
+            emptyMap()
+        ).get(taskId00);
+        restoringTask.setChangelogOffsets(changelogOffsets);
+
+        assertThat(taskManager.getTaskOffsetSums(), is(expectedOffsetSums));
+    }
+
+    @Test
+    public void shouldComputeOffsetSumForUnassignedLockableTask() throws IOException {
+        final Map<TopicPartition, Long> changelogOffsets = mkMap(
+            mkEntry(new TopicPartition("changelog", 0), 5L),
+            mkEntry(new TopicPartition("changelog", 1), 10L)
+        );
+        final Map<TaskId, Long> expectedOffsetSums = mkMap(mkEntry(taskId00, 15L));
+
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        writeCheckpointFile(taskId00, changelogOffsets);
+
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
+
+        assertThat(taskManager.getTaskOffsetSums(), is(expectedOffsetSums));
+    }
+
+    @Test
+    public void shouldNotReportOffsetSumsForUnlockableTask() throws IOException {
+        expectLockFailedFor(taskId00);
+        makeTaskFolders("0_0");
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
+
+        assertTrue(taskManager.getTaskOffsetSums().isEmpty());
+    }
+
+    @Test
+    public void shouldNotReportOffsetSumsForUnassignedTaskWithoutCheckpoint() throws IOException {
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        expect(stateDirectory.checkpointFileFor(taskId00)).andReturn(getCheckpointFile(taskId00));
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
+
+        assertTrue(taskManager.getTaskOffsetSums().isEmpty());
+    }
+
+    @Test
+    public void shouldSkipInvalidOffsetsWhenComputingOffsetSum() throws IOException {
+        final Map<TopicPartition, Long> changelogOffsets = mkMap(
+            mkEntry(new TopicPartition("changelog", 1), -1L)
+        );
+        final Map<TaskId, Long> expectedOffsetSums = mkMap(mkEntry(taskId00, 0L));
+
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        writeCheckpointFile(taskId00, changelogOffsets);
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
 
-        replay(activeTaskCreator, stateDirectory);
+        assertThat(taskManager.getTaskOffsetSums(), is(expectedOffsetSums));
+    }
 
-        final Map<TaskId, Long> taskOffsetSums = taskManager.getTaskOffsetSums();
+    @Test
+    public void shouldOffsetSumOverflowToLongMaxValue() throws IOException {
```

Comment:
prop: Rename to `shouldPinOffsetToLongMaxValueInCaseOfOverflow`

Reply:
Wow yeah that original test name made no sense, thanks for the prop